mono/metadata/w32process-unix.c

```diff
@@ -1360,6 +1329,35 @@ switch_dir_separators (char *path)
 
 MONO_SIGNAL_HANDLER_FUNC (static, mono_sigchld_signal_handler, (int _dummy, siginfo_t *info, void *context))
 {
+	/*
+	 * Don't want to do any complicated processing here so just wake up the finalizer thread which will call
+	 * mono_w32process_signal_finished ().
+	 */
+	mono_gc_finalize_notify ();
```

Comment:
We should document that mono_gc_finalize_notify  mono_coop_sem_post are async safe.
We should save/restore errno around the signal handler, we've ran into issues when the handler changing it.